src/module.c

```diff
@@ -121,23 +124,45 @@ int MGraph_Query(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     clock_t end;
 
     const char *graphName;
-    const char *query;
-    RMUtil_ParseArgs(argv, argc, 1, "cc", &graphName, &query);
+    const char *query_str;
+    RMUtil_ParseArgs(argv, argc, 1, "cc", &graphName, &query_str);
 
     /* Parse query, get AST. */
     char *errMsg = NULL;
 
-    AST_QueryExpressionNode* ast;
-
-    ast = ParseQuery(query, strlen(query), &errMsg);
+    AST_Query *ast = ParseQuery(query_str, strlen(query_str), &errMsg);
 
     if (!ast) {
         RedisModule_Log(ctx, "debug", "Error parsing query: %s", errMsg);
         RedisModule_ReplyWithError(ctx, errMsg);
         free(errMsg);
         return REDISMODULE_OK;
     }
-    
+
+    if (ast->indexNode != NULL) {
```

Comment:
Please create a separate function within module.c to hold this logic:
Something resembling switch(ast->indexNode->operation){
case CREATE_INDEX:
_create_index(); ...
}